include/conf/load.hpp

```diff
@@ -37,17 +38,50 @@ namespace color_coded
       }
     }
 
-    inline args_t load(std::string const &file, std::string const &filetype)
+    inline args_t load_compilation_database(std::string const &file, std::string filename)
     {
-      if(file.empty())
-      { return defaults(filetype); }
+      std::string error;
+      auto const database_ptr(::clang::tooling::JSONCompilationDatabase::loadFromFile(file, error));
+      if(!database_ptr)
+      { return {}; }
+
+      std::vector<fs::path> files{filename};
+      auto const ext(fs::path(filename).extension());
+      if(ext == ".h" || ext == ".hpp")
+      {
+        files.emplace_back(fs::path(filename).replace_extension("c"));
+        files.emplace_back(fs::path(filename).replace_extension("cpp"));
+      }
```

Comment:
This sounds like a reasonable thing we should do. Maybe I can find the time to implement it this weekend.